### PR TITLE
Non-bitflag slot defines removal

### PR DESCRIPTION
**Pull request by Sinestia**
## About The Pull Request
This ports 
https://github.com/tgstation/tgstation/pull/47742

also something on the Big Man's list

## Why It's Good For The Game
 Standardization, simplicity, and enhanced compatibility as well. 

prooobably warrants a test merge, albeit it functions well for me on my private server. 
## Changelog
:cl:Froststahr
code: Non-bitflag slot defines gone, slot defines are now bitshifts
/:cl:

### CODE REVIEW

**Comment by Sinestia** [comment]
Uh submitting it through vscode didn't give the opportunity to fix desc ONE SEC

**Comment by utilbird** [comment]
godspeed, beebot

**Comment by PowerfulBacon** [comment]
whats the point though? Are there any situations in which the usage of bitflags is better for slots (since I can't really see something needing to have multiple slot flags at once)

File: code/modules/mob/living/carbon/update_icons.dm [hunk 1]
```diff
@@ -114,12 +114,12 @@
 	if(!get_bodypart(BODY_ZONE_HEAD)) //Decapitated
 		return
 
-	if(client && hud_used && hud_used.inv_slots[SLOT_WEAR_MASK])
-		var/atom/movable/screen/inventory/inv = hud_used.inv_slots[SLOT_WEAR_MASK]
+	if(client && hud_used && hud_used.inv_slots[ITEM_SLOT_MASK])
```
Comment (park66665) [inline code/modules/mob/living/carbon/update_icons.dm]:
```suggestion
	if(client && hud_used && hud_used.inv_slots[TOBITSHIFT(ITEM_SLOT_MASK) + 1])
```

File: code/modules/mob/living/carbon/update_icons.dm [hunk 2]
```diff
@@ -128,12 +128,12 @@
 /mob/living/carbon/update_inv_neck()
 	remove_overlay(NECK_LAYER)
 
-	if(client && hud_used && hud_used.inv_slots[SLOT_NECK])
-		var/atom/movable/screen/inventory/inv = hud_used.inv_slots[SLOT_NECK]
+	if(client && hud_used && hud_used.inv_slots[ITEM_SLOT_NECK])
```
Comment (park66665) [inline code/modules/mob/living/carbon/update_icons.dm]:
```suggestion
	if(client && hud_used && hud_used.inv_slots[TOBITSHIFT(ITEM_SLOT_NECK) + 1])
```

File: code/modules/mob/living/carbon/update_icons.dm [hunk 3]
```diff
@@ -142,8 +142,8 @@
 /mob/living/carbon/update_inv_back()
 	remove_overlay(BACK_LAYER)
 
-	if(client && hud_used && hud_used.inv_slots[SLOT_BACK])
-		var/atom/movable/screen/inventory/inv = hud_used.inv_slots[SLOT_BACK]
+	if(client && hud_used && hud_used.inv_slots[ITEM_SLOT_BACK])
```
Comment (park66665) [inline code/modules/mob/living/carbon/update_icons.dm]:
```suggestion
	if(client && hud_used && hud_used.inv_slots[TOBITSHIFT(ITEM_SLOT_BACK) + 1])
```

File: code/modules/mob/living/carbon/update_icons.dm [hunk 4]
```diff
@@ -158,8 +158,8 @@
 	if(!get_bodypart(BODY_ZONE_HEAD)) //Decapitated
 		return
 
-	if(client && hud_used && hud_used.inv_slots[SLOT_BACK])
-		var/atom/movable/screen/inventory/inv = hud_used.inv_slots[SLOT_HEAD]
+	if(client && hud_used && hud_used.inv_slots[ITEM_SLOT_BACK])
```
Comment (park66665) [inline code/modules/mob/living/carbon/update_icons.dm]:
```suggestion
	if(client && hud_used && hud_used.inv_slots[TOBITSHIFT(ITEM_SLOT_HEAD) + 1])
```

File: code/modules/events/wizard/curseditems.dm [hunk 5]
```diff
@@ -17,23 +17,23 @@
...
-			loadout[SLOT_GLOVES] = /obj/item/clothing/gloves/boxing
+			loadout[ITEM_SLOT_MASK] = /obj/item/clothing/mask/luchador
+			loadout[ITEM_SLOT_GLOVES] = /obj/item/clothing/gloves/boxing
 			ruins_spaceworthiness = 1
 		if("voicemodulators")
-			loadout[SLOT_WEAR_MASK] = /obj/item/clothing/mask/chameleon
+			loadout[ITEM_SLOT_MASK] = /obj/item/clothing/mask/chameleon
 		if("catgirls2015")
-			loadout[SLOT_HEAD] = /obj/item/clothing/head/kitty
+			loadout[ITEM_SLOT_HEAD] = /obj/item/clothing/head/kitty
```
Comment (park66665) [inline code/modules/events/wizard/curseditems.dm]: You should apply `TOBITSHIFT` here.

**Review by park66665 — CHANGES_REQUESTED** [review]
We can reap the benefits of having a bitflag only defines for slots, but that is a PR for another time.

**Comment by Sinestia** [comment]
a pity i had to webedit again -- im basically clueless on resolving merge conflicts in an external tool

couldn't test these new changes yet

**Comment by park66665** [comment]
#3558 will be merged soon(tm) so I am going to wait before another review (or pestering folks with perms to TM this)

**Comment by utilbird** [comment]
#3558 has been merged

**Comment by Sinestia** [comment]
I will have my revenge when I break a bunch of people's PRs in return

**Comment by Sinestia** [comment]
alright there we go im not a klutz i promis

**Review by utilbird — APPROVED** [review]
need ike's tick

**Pull request merged**